core/modules/dashboard/dashboard.module

```diff
@@ -13,11 +13,21 @@ function dashboard_menu() {
   // Check if any dashboard is present and is not disabled.
   $dashboard_layouts = layout_load_multiple_by_path('admin/dashboard');
   $dashboard_layout = array_pop($dashboard_layouts);
-  if ($dashboard_layout->hasContexts(array('dashboard')) && !$dashboard_layout->disabled) {
+
+  if (!$dashboard_layout->disabled && $dashboard_layout->hasContexts(array('dashboard'))) {
```

Comment:
If you wonder why I inverted this, it's just because the second condition cost less than the first one.